src/Symfony/Bundle/FrameworkBundle/Command/WorkerRunCommand.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Symfony\Bundle\FrameworkBundle\Command;
+
+use Symfony\Component\Amqp\Worker\ConfigurableLoopInterface;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class WorkerRunCommand extends ContainerAwareCommand
+{
+    protected function configure()
+    {
+        $this
+            ->setName('worker:run')
+            ->setDescription('Run a worker')
+            ->setDefinition(array(
+                new InputArgument('worker', InputArgument::REQUIRED, 'The worker'),
+                new InputOption('name', null, InputOption::VALUE_REQUIRED, 'A name, useful for stats/monitoring. Defaults to worker name.'),
+            ))
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $loop = $this->getLoop($input);
+
+        $loopName = $input->getOption('name') ?: $loop->getName();
+
+        if ($loop instanceof ConfigurableLoopInterface) {
+            $loop->setName($loopName);
+        }
+
+        $processName = sprintf('%s_%s', $this->getContainer()->getParameter('worker.cli_title_prefix'), $loopName);
+
+        // On OSX, it may raise an error:
+        // Warning: cli_set_process_title(): cli_set_process_title had an error: Not initialized correctly
+        @cli_set_process_title($processName);
+
+        pcntl_signal(SIGTERM, function () use ($loop) {
```

Comment:
IMO pcnt related code should not be hard coded or mixed with the core code. 

One may want to run it on Windows machine where pcntl is not available or use [another extension to handle signals](https://github.com/rstgroup/php-signal-handler). 

There should be a way to disable\replace this logic. 

That's why I suggest to move it to a kind of extension. We were [able to achieve](https://github.com/php-enqueue/enqueue-dev/blob/master/pkg/enqueue/Consumption/Extension/SignalExtension.php) it in our project.